Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -142,9 +142,14 @@ private void OnPlayerJobAssigned(RulePlayerJobsAssignedEvent ev)
         var query = QueryActiveRules();
         while (query.MoveNext(out _, out var comp, out _))
         {
-            _antagSelection.EligiblePlayers(comp.RevPrototypeId, comp.MaxHeadRevs, comp.PlayersPerHeadRev, comp.HeadRevStartSound,
+            _antagSelection.EligiblePlayers(comp.HeadRevPrototypeId, comp.MaxHeadRevs, comp.PlayersPerHeadRev, comp.HeadRevStartSound,
                 "head-rev-role-greeting", "#5e9cff", out var chosen);
-            GiveHeadRev(chosen, comp.RevPrototypeId, comp);
+            if (!chosen.Any())
+                GiveHeadRev(chosen, comp.RevPrototypeId, comp);
+            else
+            {
+                _chatManager.SendAdminAnnouncement(Loc.GetString("rev-no-heads"));
```

Comment:
is there not a thing to cancel the gamemode and fallback to another, which could be used here

Reply:
There is but by now the game mode has already started so it's not possible. 

Talked in the dev channel about this, tldr is the check for if people have actually enrolled is to do it before the round begins. However thats not possible here since we need to check for if the job is antag allowed. 

I was informed that time why we do random player selection and its for this exact reason... theres no other way.

Theres no other way with the current system and im most likely not smart enough to refactor RNG job selection to expose if the player is allowed to be antag before starting the game.

Reply:
fair enough then